frontends/common/constantFolding.cpp

```diff
@@ -237,6 +237,38 @@ const IR::Node *DoConstantFolding::preorder(IR::ArrayIndex *e) {
     return e;
 }
 
+// Returns true if the given expression is of the form "some_table.apply().action_run."
+bool isActionRun(const IR::Expression *e, const ReferenceMap *refMap) {
```

Comment:
Make this a private member function or wrap it in a namespace to avoid possible future ODR complaints with unity builds?

Reply:
can't it just be `static`?

Reply:
Iirc this will not help with unity builds once two static definitions are pulled together into the same compilation unit. These name clashes do not occur that often though, so it is a minor problem, really.